### ISSUE TITLE
Inference for Source Separation

### ISSUE THREAD

**Issue by bryant0918** [s3prl/s3prl#382]
I've been able to prepare the dataset and run the evaluation for source separation following the documentation [here](https://github.com/s3prl/s3prl/blob/master/s3prl/downstream/docs/superb.md#ss-source-separation) using the upstream model wavlm and the downstream separation_stft. This returns an sr_srdi, pesq, and stoi score on the test set.

However, I would like to get an inference with my own data and not just evaluate. Is there a way to use this downstream to input an audio file and output two or more audio files of the separated speech?

I noticed there is no function for 'inference' in the separation_stft DownstreamExpert.

**Comment by bryant0918** [comment]
I've managed to save some of the output audio files with the evaluation, but it has poor quality. It doesn't seem to be separating voices at all and it just makes everything sound like a robot. Attached is my mixed audio file from LibriMix and the corresponding prediction files. 

I am using the WavLM pretrained model. I'm wondering if I just need to train the model more, or if there is something wrong with my code. My process is as follows:

1. Load the wav file with Librosa
2. Apply the transpose of the STFT with librosa
3. Convert it into a Tensor object and pad the sequence
4. Apply the upstream model (WavLM)
5. Use the Featurizer model to extract my hidden state and call the forward function
6. Use the downstream model (SepRNN from PITNet Github) to create my list of masks
7. Take the element wise product of the masks and the STFTs using torch.squeeze then transpose
8. Take the inverse STFT with librosa
9. write my output audio file with Soundfile. 

[Libri2.zip](https://github.com/s3prl/s3prl/files/9252950/Libri2.zip)

**Comment by mixiazhiyang** [comment]
> 
have you solved this problem after training longer?

**Comment by leo19941227** [comment]
Hi,

Since our speech enhancement/separation pipelines follow the Kaldi style data preparation, so I believe you can simply prepare the testing kaldi data dir and **override** the config with `-o` option the doing evaluation `-m evaluate`.
